src/main/java/org/wso2/carbon/connector/operations/RedisServer.java

```diff
@@ -90,6 +92,17 @@ public RedisServer(MessageContext messageContext) {
      * @return Jedis instance
      */
     private Jedis createJedis() {
+        String connectionURIProp = (String) messageContext.getProperty(RedisConstants.CONNECTION_URI);
+        if (connectionURIProp != null) {
+            try {
+                URI connectionURI = new URI(connectionURIProp);
+                return new Jedis(new JedisShardInfo(connectionURI));
+            } catch (URISyntaxException e) {
```

Comment:
How do we use weight, SOTimeout and connection timeout with connection URL?

Reply:
According to [1], the weight is set to the default weight, which is 1 and there is no setter method or appropriate constructor to set the weight along with the connection URI. 

[1] - https://github.com/redis/jedis/blob/f24a817987fe1990a9ff65b31dba7fa4b44969e6/src/main/java/redis/clients/jedis/JedisShardInfo.java#L165

Reply:
Called the setter methods to set SOTimeout and connection timeout.